routers/repo/branch.go

```diff
@@ -5,28 +5,98 @@
 package repo
 
 import (
+	"code.gitea.io/git"
 	"code.gitea.io/gitea/modules/base"
 	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/log"
 )
 
 const (
-	tplBranch base.TplName = "repo/branch"
+	tplBranch base.TplName = "repo/branch/list"
 )
 
+// Branch contains the branch information
+type Branch struct {
+	Name   string
+	Commit *git.Commit
+}
+
 // Branches render repository branch page
 func Branches(ctx *context.Context) {
 	ctx.Data["Title"] = "Branches"
 	ctx.Data["IsRepoToolbarBranches"] = true
+	ctx.Data["DefaultBranch"] = ctx.Repo.Repository.DefaultBranch
+	ctx.Data["IsBranchDeletable"] = ctx.Repo.IsWriter()
+
+	ctx.Data["Branches"] = loadBranches(ctx)
+	ctx.HTML(200, tplBranch)
+}
+
+// DeleteBranchPost responses for delete merged branch
+func DeleteBranchPost(ctx *context.Context) {
+	branchName := ctx.Query("name")
+
+	defer func() {
+		redirectTo := ctx.Query("redirect_to")
```

Comment:
Do you really need redirect_to from post? It is known that it will allways be branch overview page